operator_ui/src/pages/Jobs/New.tsx

```diff
@@ -0,0 +1,270 @@
+  getJobSpecFormat,
+  stringifyJobSpec,
+  isJson,
+  isToml,
+} from 'utils/jobSpec'
+import { ApiResponse } from '@chainlink/json-api-client'
+import Button from 'components/Button'
+import * as api from 'api'
+import { useDispatch } from 'react-redux'
+import {
+  OcrJobSpecRequest,
+  OcrJobSpec,
+  JobSpecRequest,
+} from 'core/store/models'
+import { JobSpec } from 'core/store/presenters'
+import BaseLink from 'components/BaseLink'
+import ErrorMessage from 'components/Notifications/DefaultError'
+import { notifySuccess, notifyError } from 'actionCreators'
+import * as storage from '@chainlink/local-storage'
+import Content from 'components/Content'
+import {
+  TextField,
+  Grid,
+  Card,
+  CardContent,
+  FormControlLabel,
+  FormControl,
+  FormLabel,
+  RadioGroup,
+  Divider,
+  CardHeader,
+  CircularProgress,
+} from '@material-ui/core'
+import {
+  createStyles,
+  withStyles,
+  WithStyles,
+  Theme,
+} from '@material-ui/core/styles'
+import { useLocation, useHistory } from 'react-router-dom'
+import { setPersistJobSpec, getPersistJobSpec } from 'utils/storage'
+
+const jobSpecFormatList = [JobSpecFormats.JSON, JobSpecFormats.TOML]
+export const SELECTED_FORMAT = 'persistSpecFormat'
+
+const styles = (theme: Theme) =>
+  createStyles({
+    card: {
+      padding: theme.spacing.unit,
+      marginBottom: theme.spacing.unit * 3,
+    },
+    loader: {
+      position: 'absolute',
+    },
+  })
+
+const SuccessNotification = ({ id }: { id: string }) => (
+  <>
+    Successfully created job{' '}
+    <BaseLink id="created-job" href={`/jobs/${id}`}>
+      {id}
+    </BaseLink>
+  </>
+)
+
+export function validate({
+  format,
+  value,
+}: {
+  format: JobSpecFormats
+  value: string
+}) {
+  if (format === JobSpecFormats.JSON && isJson({ value })) {
+    return true
+  } else if (format === JobSpecFormats.TOML && isToml({ value })) {
+    return true
+  } else {
+    return false
+  }
+}
+
+function apiCall({
+  format,
+  value,
+}: {
+  format: JobSpecFormats
+  value: string
+}): Promise<ApiResponse<JobSpec | OcrJobSpec>> {
+  if (format === JobSpecFormats.JSON) {
+    const definition: JobSpecRequest = JSON.parse(value)
+    return api.v2.specs.createJobSpec(definition)
+  }
+
+  if (format === JobSpecFormats.TOML) {
+    const definition: OcrJobSpecRequest = { toml: value }
+    return api.v2.ocrSpecs.createJobSpec(definition)
+  }
+
+  return Promise.reject('Invalid format')
+}
+
+function getInitialValues({
+  query,
+}: {
+  query: string
+}): { jobSpec: string; format: JobSpecFormats } {
+  const params = new URLSearchParams(query)
+  const jobSpec = (params.get('definition') as string) || getPersistJobSpec()
+
+  const format =
+    getJobSpecFormat({
+      value: jobSpec,
+    }) ||
+    JobSpecFormats[params.get('format')?.toUpperCase() as JobSpecFormat] ||
+    (storage.get(SELECTED_FORMAT) as JobSpecFormat) ||
+    JobSpecFormats.JSON
+
+  return {
+    jobSpec: stringifyJobSpec({ value: jobSpec || '' }),
```

Comment:
Wouldn't this blow up if the job spec is TOML?

Reply:
No, this is ok but `''` was redundant. Ive refactored stringifyJobSpec function to accept `format` as well so now we can stringify JSON and potentially TOML files in future (we will get to that point on adding TOML/OCR job details page with `duplicate spec` functionality)